tests/migrations/test_operations.py

```diff
@@ -1801,6 +1801,155 @@ def inner_method(models, schema_editor):
             create_old_man.state_forwards("test_books", new_state)
             create_old_man.database_forwards("test_books", editor, project_state, new_state)
 
+    def test_model_with_bigautofield(self):
+        """
+        Tests the creation of a model with BigAutoField.
+        """
+
+        def inner_method(models, schema_editor):
+            Author = models.get_model("test_author", "Author")
+            Book = models.get_model("test_book", "Book")
+            author1 = Author.objects.create(name="Hemingway")
+            Book.objects.create(title="Old Man and The Sea", author=author1)
+            Book.objects.create(id=2 ** 33, title="A farewell to arms", author=author1)
+
+            author2 = Author.objects.create(id=2 ** 33, name="Remarque")
+            Book.objects.create(title="All quiet on the western front", author=author2)
+            Book.objects.create(title="Arc de Triomphe", author=author2)
+
+        create_author = migrations.CreateModel(
+            "Author",
+            [
+                ("id", models.BigAutoField(primary_key=True)),
+                ("name", models.CharField(max_length=100)),
+            ],
+            options={},
+        )
+        create_book = migrations.CreateModel(
+            "Book",
+            [
+                ("id", models.BigAutoField(primary_key=True)),
+                ("title", models.CharField(max_length=100)),
+                ("author", models.ForeignKey(to="test_author.Author", on_delete=models.CASCADE))
+            ],
+            options={},
+        )
+        fill_data = migrations.RunPython(inner_method, inner_method)
```

Comment:
I'd omit the second "inner_method" since it's not meant as a backwards operation and not used as far as I can tell.